### PR TITLE
Added archive argument to the Sagemaker Batch Transform integration

### PR DESCRIPTION
**Pull request by YQ-Wang**
## What changes are proposed in this pull request?

An argument _archive_ was added to the Sagemaker Batch Transform APIs and CLIs to specify whether to keep the Sagemaker model and model artifacts once the transform job finishes.

## How is this patch tested?

Manually deployed a Sagemaker Batch job with both API and CLI.
Ran all the test cases.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [x] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by YQ-Wang** [comment]
Hey @dbczumar, I added a small change ([3e4f44f](https://github.com/mlflow/mlflow/pull/4589/commits/3e4f44f2e33ae146adc653be39582ff4371dbdfe) & [0bc8cf8](https://github.com/mlflow/mlflow/pull/4589/commits/0bc8cf8e2b458e662d119008274cd975438ba454)) that amended to the previous PR. Here a newly added argument _archive_ can be used to specify whether to keep the Sagemaker model and model artifacts once the batch transform job finishes. If set _archive=False_, the following model and model artifacts will be cleaned.
Screen Shot 2021-07-22 at 1 30 57 AM
Screen Shot 2021-07-22 at 1 31 41 AM

Thank you for reviewing this PR.

**Comment by dbczumar** [comment]
@YQ-Wang Happy to take a look once merge conflicts have been resolved!

**Comment by YQ-Wang** [comment]
Hi @dbczumar, I have resolved the conflicts. Thank you for taking a look!

File: mlflow/sagemaker/__init__.py
Comment (dbczumar) [inline mlflow/sagemaker/__init__.py:605]: @YQ-Wang Can we clarify that the resources are archived after the job terminates?

**Review by dbczumar — APPROVED** [review]
@YQ-Wang LGTM once https://github.com/mlflow/mlflow/pull/4589/files#r677182458 has been addressed. Thank you for your contribution! I'll restart the CI here.

**Comment by YQ-Wang** [comment]
Hey @dbczumar, I have updated the doc to clarify the finished transform job. Thank you for the code review!

**Comment by YQ-Wang** [comment]
@dbczumar, seems some test cases keep queuing up and canceling. I noticed there was a merged commit to allow concurrent Github actions. Let me know if there is more work that needs to be done for that area.

**Comment by dbczumar** [comment]
@YQ-Wang We're working with GitHub support to resolve the issue. We'll make sure to get your PR merged shortly.

**Comment by dbczumar** [comment]
Woohoo! Looks like CI passed. Merging! Thanks @YQ-Wang !

**Comment by YQ-Wang** [comment]
Thanks @dbczumar! I will keep looking at this project and making contributions as needed.

**Comment by dbczumar** [comment]
Fantastic! We appreciate it :D

**Pull request merged**
